pages/_document.js

```diff
@@ -1,6 +1,17 @@
+/* eslint-env node */
+
 import React from 'react'
 import Document, { Head, Main, NextScript } from 'next/document'
 import { ServerStyleSheet } from 'styled-components'
+import * as Sentry from '@sentry/browser'
+
+process.on('unhandledRejection', err => {
+  Sentry.captureException(err)
+})
+
+process.on('uncaughtException', err => {
+  Sentry.captureException(err)
+})
```

Comment:
Why are these needed if Sentry supposedly has automated handling of uncaught exceptions and unhandled rejections? See https://docs.sentry.io/platforms/javascript/#automatically-capturing-errors

Reply:
It's for server side errors in node.js, Sentry will automatically log errors in browser, but not unhandled errors in SSR.

Reply:
Got it, thanks.

Would these also be needed separately for server.js or other build-time code? e.g. sidebar.js

Reply:
This catches all code in the actual pages including `sidebar.js`, but I think we are now missing errors from server.js because they happens before we call `_document.js`.

Reply:
Yes some server code runs before the Next app and partially includes sidebar.js code I think. That's what I was wondering about. But it's probably not a big deal? Only runs when the server is being initialized, probably only when new versions are deployed to Heroku.

Reply:
That's pretty much my thoughts as well. And also the fact that we are planning to migrate to Gatsby and it didn't have a server.  So I decided to leave it as is for now.